src/components/InputField.tsx

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable react/require-default-props */
+import React, { useState } from "react";
+
+import * as Styled from "./InputField.styled";
+
+interface InputFieldProps {
+  id: string;
+  type: string;
+  labelValue: string;
+  inputValue: string;
+  inputSize?: string;
+  isRequired?: boolean;
+  onChange: any;
+  hasLabel?: boolean;
+  socialIcon?: string;
+}
+
+function InputField({
+  id,
+  type,
+  labelValue,
+  inputValue,
+  inputSize,
+  onChange,
+  isRequired,
+  socialIcon,
+  hasLabel = true,
+}: InputFieldProps) {
+  const [isValid, setIsValid] = useState(true);
+
+  return (
+    <Styled.InputField
+      socialIcon={socialIcon}
+      hasLabel={hasLabel}
+      isValid={isValid}
+      type={type}
+      inputSize={inputSize}
```

Comment:
Just a suggestion, when the name + value are the same, you can also pass your props like this: `{...{ socialIcon, hasLabel, isValid, type, inputSize }}`. Whatever you prefer :)

Reply:
Nice!